zng/type.go

```diff
@@ -83,6 +77,15 @@ const (
 	IdTime     = 16
 	IdDuration = 17
 	IdAny      = 18
+
+	IdTypeDef = 23
```

Comment:
I asked about 23 in the past and the answer involved "universal constant". More importantly though are we sure we don't want to leave room for more than 5 additional primitive types?

(any discussion doesn't have to hold up this PR; the spec is Alpha)

Reply:
I was waiting till the dust settled on the spec.  e.g., I think it might make sense to get rid of many of the integer types.  e.g., avro has only int and long

Reply:
Can we make sure we keep something that's an unsigned 64-bit integer? It's a small victory but I always liked needling the JSON-centric solutions for not being able to ingest the whole range of possible Zeek `count` values.